portal-impl/src/com/liferay/portal/util/PortalImpl.java

```diff
@@ -923,7 +923,14 @@ public String escapeRedirect(String url) {
 				else if (allowedDomain.equals(domain)) {
 					return url;
 				}
-				else if (allowedDomain.equals("PORTAL_DOMAIN")) {
+				else if (allowedDomain.equals("PORTAL_DOMAINS")) {
+					if (Validator.isNotNull(
+							VirtualHostLocalServiceUtil.fetchVirtualHost(
+								domain))) {
+
+						return url;
+					}
+
 					ServiceContext serviceContext =
 						ServiceContextThreadLocal.getServiceContext();
```

Comment:
Here we keep accepting the current portal domain, in case the user still needs to navigate through a non stored portal virtual host. (redirects to it from a different portal domain won't work, unless it's added to the domain redirect list).